pandas/tests/indexing/test_loc.py

```diff
@@ -1285,6 +1286,14 @@ def test_loc_setitem_datetime_keys_cast(self):
             expected = DataFrame({"one": [100.0, 200.0]}, index=[dt1, dt2])
             tm.assert_frame_equal(df, expected)
 
+    def test_loc_setitem_categorical_column_retains_dtype(self):
```

Comment:
can you add the ordered fixture (just include it in the arg list and pass to Categorical), it should just work. ping on green.